controllers/apim/ratelimitpolicy_controller.go

```diff
@@ -108,61 +105,128 @@ func (r *RateLimitPolicyReconciler) Reconcile(eventCtx context.Context, req ctrl
 		return ctrl.Result{}, err
 	}
 
-	for _, networkingRef := range rlp.Spec.NetworkingRef {
-		switch networkingRef.Type {
-		case apimv1alpha1.NetworkingRefTypeHR:
-			logger.Info("HTTPRoute is not implemented yet") // TODO(rahulanand16nov)
-			return ctrl.Result{}, nil
-		case apimv1alpha1.NetworkingRefTypeVS:
-			vsNamespacedName := client.ObjectKey{
-				Namespace: rlp.Namespace, // VirtualService lookup is limited to RLP's namespace
-				Name:      networkingRef.Name,
-			}
+	// Operation specific annotations must be removed if they were present
+	updateRequired := false
+	// check for delete operation for virtualservice
+	if vsName, present := rlp.Annotations[KuadrantDeleteVSAnnotation]; present {
```

Comment:
can you explain the use case for this annotation?  Around deletion of the VS, we can add a finalizer to the virtual service and then if it is deleted it wont be removed until we remove the finalizer. This will give us chance to clean up any associated Istio resources

Reply:
@eguzki perhaps you can help @rahulanand16nov with this ?

Reply:
`delete` operation is used even when the annotation is removed from the VS.